docs/tutorial.md

```diff
@@ -0,0 +1,227 @@
+
+- `bot.on(eventName, listener)`
+  Execute the `listener` function for each time the event named `eventName` triggered.
+- `bot.once(eventName, listener)`
+  Execute the `listener` function, the first time the event named `eventName` triggered.
+- `bot.removeListener(eventName, listener)`
+  Removes the specified `listener` for the event named `eventName`.
+
+Not only bot object, [`Chest`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerchest), [`Furnace`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerfurnace), [`Dispenser`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerdispenser), [`EnchantmentTable`](http://mineflayer.prismarine.js.org/#/api?id=mineflayerenchantmenttable), [`Villager`](http://mineflayer.prismarine.js.org/#/api?id=mineflayervillager) object also have its own events!
+
+Examples :
+
+#### Echo Bot
+
+Here we're listening for [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and resend the message.
+[`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event is only emitted when a player chats publicly.
+
+```js
+function echo(username, message) {
+  if (username === bot.username) return;
+  bot.chat(message);
+}
+
+bot.on("chat", echo); // Only emitted when a player chats publicly.
+```
+
+`bot.on('chat', echo)` will listen for [`chat`](http://mineflayer.prismarine.js.org/#/api?id=quotchatquot-username-message-translate-jsonmsg-matches) event, and execute `echo` function with arguments `username`, `message`, `translate`, `jsonMsg`, and `matches` when the event emitted.
+
+#### Welcome Bot
+
+Here we're listening for [`playerJoined`](http://mineflayer.prismarine.js.org/#/api?id=quotplayerjoinedquot-player) event.
+And sending welcome message to chat.
+
+```js
+function joined(player) {
+  bot.chat(`Welcome ${player}! :D`);
+}
+
+bot.on("playerJoined", joined);
+```
+
+### Callbacks
+
+Callback is a function as argument to the main function, when the main function finish or got an error then the callback executed.
+In [the API page](http://mineflayer.prismarine.js.org/#/api) callback abbreviated as `cb` or `callback`.
+Callback usually takes parameter called `err` as the first and the second (if specified) is a value that is passed to the callback,
+if `err` is not `null` then error must be happened in the main function.
+
+Examples :
+
+#### Craft Stick Bot
+
+Here we're crafting oak logs into sticks.
+Craft oak logs into oak wood planks, wait for crafting finish. Then craft the oak wood planks into sticks.
+
+Incorect aproach ❌ :
+
+```js
+const plankRecipe = bot.recipesFor(5); // 5 ID for Oak Wood Planks
+bot.craft(plankRecipe, 1); // ❌
+const stickRecipe = bot.recipesFor(280); // 280 ID for Stick
+bot.craft(plankRecipe, 1); // ❌
+// You don't wanna start crafting sticks, before crafting logs into planks finished.
```

Comment:
That comment still doesn't say anything useful. It should be explained why it is an incorrect approach. Say why is it incorrect, what may happen, what leads to the crash and why.

Reply:
Alrignt. i'll put more explanation there